lib/route/route_obj.c

```diff
@@ -1339,6 +1361,10 @@ int rtnl_route_parse(struct nlmsghdr *nlh, struct rtnl_route **result)
 			return err;
 	}
 
+	if (tb[RTA_NH_ID]) {
+		rtnl_route_set_nhid(route, nla_get_u32(tb[RTA_NH_ID]));
```

Comment:
I suspect, that kernel would never send RTA_NH_ID set to zero? Then this code is fine.

Otherwise, what does it mean, if kernel would send RTA_NH_ID set to zero? I think, to libnl that should mean that `ROUTE_ATTR_NHID` is *not* set, and we should skip calling `rtnl_route_set_nhid()`.

But as I suspect that can never happen (kernel wouldn't send this), it's fine as is.